libvast/vast/table_slice.hpp

```diff
@@ -43,11 +46,20 @@ class table_slice : public caf::ref_counted {
 
   // -- properties -------------------------------------------------------------
 
-  /// @returns The table layout.
+  /// @returns the table layout.
   inline const record_type& layout() const noexcept {
     return layout_;
   }
 
+  /// @returns the content of a row wrapped into an event.
+  caf::optional<event> row_to_event(size_type row, size_type first_column = 0u,
+                                    size_type num_columns = npos) const;
+
+  std::vector<event> rows_to_events(size_type first_row = 0u,
+                                    size_type num_rows = npos,
+                                    size_type first_column = 0u,
+                                    size_type num_columns = npos) const;
+
```

Comment:
Returning `value` instead of `event` makes more sense here, since a `table_slice` doesn't know about the additional meta data that `event` provides.

Reply:
What about the name? `rows_to_values` instead?